File: examples/factorial.c
#include <stdio.h>
long int fact(int n);

int main() {
	int n;
	printf("Ведите n: ");
	scanf("%d", &n);
	printf("%d! = %ld\n", n, fact(n));
}

long int fact(int n) {
	if (n >= 1)
		return n*fact(n-1);
	else
		return 1;
}

// max: 20! = 2432902008176640000
// 15! = 1307674368000
// 13! = 6227020800

File: examples/factorial.rs
use std::io;

fn fact(x: i64) -> i64 {
    if x >= 1 {
    	return x*fact(x-1);
    } 
    else {
    	return 1;
    }
}	

fn main() {
	println!("Введите n:");

	let mut n = String::new();

	io::stdin().read_line(&mut n)
	    .ok()
	    .expect("Не удалось прочитать строку");

	let n: i64 = n.trim().parse()
	    .ok()
	    .expect("Введите число!");
	
	println!("{}! = {}", n, fact(n));
}

// max: 20! = 2432902008176640000